src/main/java/ca/utoronto/fitbook/adapter/persistence/firebase/ExerciseFirebaseRepository.java

```diff
@@ -102,4 +100,40 @@ public List<Exercise> loadExerciseList(List<String> exerciseIds) throws EntityNo
         }
         return exerciseList;
     }
+
+    /**
+     *
+     * @return A list of all exercises in the database
+     */
+    @Override
+    public ArrayList<Exercise> loadAllExercises() {
+        Iterable<DocumentReference> future = firestore.collection(COLLECTION_NAME).listDocuments();
+        ArrayList<Exercise> exercises = new ArrayList<>();
+        for (DocumentReference reference : future) {
+            ApiFuture<DocumentSnapshot> snapshot = reference.get();
+            try {
+                exercises.add(getExerciseHelper(snapshot.get()));
```

Comment:
You should be able to utilize the `.toObject` method if you `.get()` the future snapshot for a DocumentSnapshot.